docs: add documentation for ngNonBindable

ajitsinghkaler opened the pull request:
Ng non-bindable documentation was not present added documentation for ng non-bindable

@kapunahelewong was adding this documentation in #28577 asked me to take over for her as she doesn't have time to complete

Fixes #28577 #19497

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
no ngNonBindable documentation

Issue Number: #19497


## What is the new behavior?
added ngNonBindable documentation

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

ajitsinghkaler commented:
This needs engineering review and finalizations

aikidave commented on aio/content/guide/attribute-directives.md:
```diff
@@ -303,14 +303,35 @@ and fall back to "violet" as the default color.
+and will not display `2`. This is helpful when you want to render code in the browser.
+
+When you apply `ngNonBindable` to an element, it stops any binding starting at that element, including child elements. However, `ngNonBindable` still allows
+directives to work to the element where you apply`ngNonBindable`. For example, in the following, the `appHighlight` directive will still be active but Angular will not evaluate the expression `{{ 1 +1 }}`.
```
```suggestion
directives to work to the element where you apply`ngNonBindable`. In the following example, the `appHighlight` directive will still be active but Angular will not evaluate the expression `{{ 1 +1 }}`.
```

aikidave commented on aio/content/guide/attribute-directives.md:
```diff
@@ -303,14 +303,35 @@ and fall back to "violet" as the default color.
+
+<code-example path="attribute-directives/src/app/app.component.html" linenums="false" header="src/app/app.component.html" region="ngNonBindable-with-directive"></code-example>
+
+Additionally, if you apply `ngNonBindable` to a parent element, interpolation and binding of any sort, such as property binding, or event binding, will be disabled for its children.
```
```suggestion
Additionally, if you apply `ngNonBindable` to a parent element, interpolation and binding of any sort, such as property binding or event binding, is disabled for its children.
```

aikithoughts reviewed (commented):
Looking good! I have offered a few minor suggestions

ajitsinghkaler commented:
@aikidave made the changes please have a look

rafaelss95 commented on aio/content/guide/attribute-directives.md:
```diff
@@ -303,14 +303,35 @@ and fall back to "violet" as the default color.
+and will not display `2`. This is helpful when you want to render code in the browser.
+
+When you apply `ngNonBindable` to an element, it stops any binding starting at that element, including child elements. However, `ngNonBindable` still allows
+directives to work to the element where you apply`ngNonBindable`. In the following example, the `appHighlight` directive will still be active but Angular will not evaluate the expression `{{ 1 +1 }}`.
```
Excuse me, If I could give my two cents here, I think there are missing whitespaces:

- After "apply" word;
- After plus sign (there are other ocurrences also in file "app.component.html");

```suggestion
directives to work to the element where you apply `ngNonBindable`. In the following example, the `appHighlight` directive will still be active but Angular will not evaluate the expression `{{ 1 + 1 }}`.
```

ajitsinghkaler replied:
Thanks will add

aikidave replied:
Good catch, @rafaelss95!

ajitsinghkaler replied:
made the changes @aikidave please have a look

rafaelss95 replied:
@ajitsinghkaler it seems like you forgot to change it in the `app.component.html` also 👀

ajitsinghkaler commented:
@mhevery can ou please have a look

ajitsinghkaler commented:
@mhevery  can you please have a look

ajitsinghkaler commented:
@mhevery can you please review this

@aikidave can you please put a tag that it needs eng input

ajitsinghkaler committed:
docs: add documentation for ngNonBindable

ngNonBindable documentation was not present, on docs site added documentation for ngNonBindable. With this template primitive, Angular won't
evaluate expressions in elements.

Fixes #28577 Fixes #19497

ajitsinghkaler commented:
@mhevery can you please review this
